src/Nox.Solution/Models/Domain/Entities/Entity.cs

```diff
@@ -69,5 +70,103 @@ internal bool ApplyDefaults()
             if (string.IsNullOrWhiteSpace(PluralName)) PluralName = Name.Pluralize();
             return true;
         }
+
+        public IEnumerable<KeyValuePair<EntityMemberType, NoxSimpleTypeDefinition>> GetAllMembers()
+        {
+            foreach (var key in Keys!)
+            {
+                yield return new (EntityMemberType.Key, key);
+            }
+            
+            if (Attributes is not null ) 
+            { 
+                foreach (var attribute in Attributes)
+                {
+                    yield return new (EntityMemberType.Attribute, attribute);
+                }
+            }
+
+            if (OwnedRelationships is not null)
+            {
+                foreach (var relationship in OwnedRelationships)
+                {
+                    if (relationship.Related.Entity?.Keys is null)
+                    {
+                        if (relationship.Relationship == EntityRelationshipType.OneOrMany || relationship.Relationship == EntityRelationshipType.ZeroOrMany)
+                        {
+                            var foreignKeyName = $"{relationship.Related.Entity!.PluralName}";
+                            var foreignKey = new NoxSimpleTypeDefinition()
+                            {
+                                Name = foreignKeyName,
+                                Description = $"A unique identifier for a {relationship.Related.Entity!.Name}.",
+                                Type = Types.NoxType.Collection,
+                                IsRequired = true,
+                                IsReadonly = true,
+                            };
+                            yield return new(EntityMemberType.OwnedRelationship, foreignKey);
+                        }
+                        else
+                        {
+                            var foreignKeyName = $"{relationship.Related.Entity!.Name}Id";
+                            var foreignKey = new NoxSimpleTypeDefinition()
+                            {
+                                Name = foreignKeyName,
+                                Description = $"A unique identifier for a {relationship.Related.Entity!.Name}.",
+                                Type = Types.NoxType.AutoNumber,
+                                IsRequired = true,
+                                IsReadonly = true,
+                            };
+                            yield return new(EntityMemberType.OwnedRelationship, foreignKey);
+                        }
+                    }
+                    else
+                    {
+                        foreach (var key in relationship.Related.Entity.Keys)
```

Comment:
The code is practically the same as in relationships below (lines 140-145) except setting the type (EntityMemberType.OwnedRelationship/EntityMemberType.Relationship), so maybe it is possible to extract a common method.